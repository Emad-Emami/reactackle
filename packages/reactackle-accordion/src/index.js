import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
