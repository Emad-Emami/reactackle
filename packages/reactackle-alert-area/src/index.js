import AlertArea from './AlertArea';

export { AlertArea };
