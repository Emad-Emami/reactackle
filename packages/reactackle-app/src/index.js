import App from './App';

export { App };
export * from './BottomRegion';
export * from './TopRegion';
export * from './MainRegion';
