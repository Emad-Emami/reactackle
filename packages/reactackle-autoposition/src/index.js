import AutoPosition, { recalculateAllAutoPosition } from './AutoPosition';

export { AutoPosition, recalculateAllAutoPosition };
