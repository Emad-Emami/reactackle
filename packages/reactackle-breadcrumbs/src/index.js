import Breadcrumbs from './Breadcrumbs';

export { Breadcrumbs };
export * from './BreadcrumbsItem';
