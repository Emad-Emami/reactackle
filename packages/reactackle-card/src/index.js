import Card from './Card';

export { Card };
export * from './CardActions';
export * from './CardAreaMain';
export * from './CardAreaSide';
export * from './CardContent';
export * from './CardHeaderPrimary';
export * from './CardHeaderSecondary';
export * from './CardMedia';
