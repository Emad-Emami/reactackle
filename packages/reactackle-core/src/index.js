// theme
// defaultTheme
// composeReactackleTheme
// defaultThemeBlueprintBroadcast
// registerDefaultComponentTheme ++
// extractThemeOrDefault ++

// styleHelpers - for combining local theme
// export everything

// styleFunctions
// export everything

// - resetStyles
// - globalStyles
// - heladerClasses

// base
// export nothing

// constants
// export everything

// mixins
// export everything

// import {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// } from './theme';

// export {
//   registerDefaultComponentTheme,
//   extractThemeOrDefault,
// };

export * from './theme';
export * from './mixins';
export * from './styleFunctions';
export * from './styleHelpers';
export * from './utils';
export * from './constants/constants';
export * from './hoc/withExternalProps';
export * from './globalStyles';
export * from './resetStyles';
export * from './Theme/Theme';
export * from './Theme/withTheme';
