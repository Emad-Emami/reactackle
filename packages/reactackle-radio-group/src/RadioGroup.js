'use strict';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { noop, getKey, isUndef } from 'reactackle-core';
import { Radio } from './Radio/Radio';
import { RadioGroupStyled } from './styles/RadioGroupStyled';

const propTypes = {
  /**
   * Default value of the RadioGroup
   */
  defaultValue: PropTypes.any,
  /**
   * Value of the RadioGroup
   */
  value: PropTypes.any,
  /**
   * Options of the RadioGroup
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.any.isRequired,
      text: PropTypes.string,
      disabled: PropTypes.bool,
    }),
  ),
  /**
   * Specify function to call on RadioGroup value change
   */
  onChange: PropTypes.func,
};
const defaultProps = {
  defaultValue: void 0,
  value: void 0,
  options: [],
  onChange: noop,
};

export default class RadioGroup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: isUndef(props.value) ? props.defaultValue : props.value,
    };

    this._handleChange = this._handleChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.value !== this.props.value && isUndef(nextProps.value)) {
      this.setState({
        value: this.props.value,
      });
    }
  }

  _getCurrentValue() {
    return isUndef(this.props.value) ? this.state.value : this.props.value;
  }

  _handleChange({ value }) {
    const currentValue = this._getCurrentValue();

    if (currentValue === value) return;

    if (isUndef(this.props.value)) {
      this.setState({
        value,
      });
    }

    this.props.onChange({
      value,
    });
  }

  _renderOptions() {
    const value = this._getCurrentValue();

    return this.props.options.map((item, k) =>
      <Radio
        {...item}
        key={getKey(item, k)}
        index={k}
        onChange={this._handleChange}
        checked={value === item.value}
      />,
    );
  }

  render() {
    const options = this._renderOptions();

    return (
      <RadioGroupStyled>
        {options}
      </RadioGroupStyled>
    );
  }
}

RadioGroup.propTypes = propTypes;
RadioGroup.defaultProps = defaultProps;
RadioGroup.displayName = 'RadioGroup';
