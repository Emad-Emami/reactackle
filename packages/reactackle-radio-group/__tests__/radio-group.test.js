'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { setProps, findByType } from 'reactackle-core';
import { RadioGroup } from '../src';

jest.mock('react-dom');

const options = [
  {
    text: 'value1',
    value: 1,
  },
];

describe('<RadioGroup/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <RadioGroup />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  it('handles onChange', () => {
    const mockChange = jest.fn();
    const component = renderer.create(
      <RadioGroup
        onChange={mockChange}
        options={options}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();

    const input = findByType(component.toJSON(), 'input');
    input.props.onChange({ target: { checked: true } });

    expect(mockChange).toBeCalledWith({ value: 1 });
    expect(component.toJSON()).toMatchSnapshot();
  });
  
  it('handles props change', () => {
    const component = renderer.create(
      <RadioGroup options={[]} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
    setProps(component, { data: [{}] });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
