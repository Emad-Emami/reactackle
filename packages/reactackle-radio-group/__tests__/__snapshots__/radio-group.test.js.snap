// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup/> handles onChange 1`] = `
<div
  className="RadioGroupStyled__RadioGroupStyled-ihhnPC eZpICI"
>
  <div
    className="RadioStyled__RadioStyled-jaalqI bXNCba"
  >
    <input
      checked={false}
      disabled={false}
      id="radio-0"
      onChange={[Function]}
      tabIndex="0"
      type="radio"
    />
    <label
      checked={false}
      className="RadioLabelStyled__RadioLabelStyled-dllnoj cYjtBt"
      disabled={false}
      htmlFor="radio-0"
    >
      value1
    </label>
  </div>
</div>
`;

exports[`<RadioGroup/> handles onChange 2`] = `
<div
  className="RadioGroupStyled__RadioGroupStyled-ihhnPC eZpICI"
>
  <div
    className="RadioStyled__RadioStyled-jaalqI bXNCba"
  >
    <input
      checked={true}
      disabled={false}
      id="radio-0"
      onChange={[Function]}
      tabIndex="0"
      type="radio"
    />
    <label
      checked={true}
      className="RadioLabelStyled__RadioLabelStyled-dllnoj kzRwPf"
      disabled={false}
      htmlFor="radio-0"
    >
      value1
    </label>
  </div>
</div>
`;

exports[`<RadioGroup/> handles props change 1`] = `
<div
  className="RadioGroupStyled__RadioGroupStyled-ihhnPC eZpICI"
/>
`;

exports[`<RadioGroup/> handles props change 2`] = `
<div
  className="RadioGroupStyled__RadioGroupStyled-ihhnPC eZpICI"
/>
`;

exports[`<RadioGroup/> renders correctly with default props 1`] = `
<div
  className="RadioGroupStyled__RadioGroupStyled-ihhnPC eZpICI"
/>
`;
